### PR TITLE
Added /smartcard-logon option to set flag. (Stripped version of #4837…

### PR DESCRIPTION
**Pull request by akallabeth**
… by @informatimago)

This pull takes the idea of #4837 and only adds the `smartcard-logon` command line option without the filtering (split to #4840).
Stripped all unused additions (Smartcard variables currently unused)

### CODE REVIEW

File: libfreerdp/core/info.c [hunk 1]
```diff
@@ -43,6 +43,67 @@ static const char* const INFO_TYPE_LOGON_STRINGS[4] =
...
+	for (i = 0; i < ARRAYSIZE(info_flags); i ++)
+	{
+		if (info_flags[i].flag & flags)
+		{
+			strcat(result, info_flags[i].label);
+			strcat(result, "|");
+		}
+	}
+
+	result[strlen(result) - 1] = '\0'; /* remove last "|" */
```
Comment (mfleisz) [inline libfreerdp/core/info.c]: Potential OOB write if result is empty (no flags specified).

File: libfreerdp/core/info.c [hunk 2]
```diff
@@ -43,6 +43,68 @@ static const char* const INFO_TYPE_LOGON_STRINGS[4] =
...
+
+	if (!result)
+		return 0;
+
+	for (i = 0; i < ARRAYSIZE(info_flags); i ++)
+	{
+		if (info_flags[i].flag & flags)
+		{
+			strcat(result, info_flags[i].label);
+			strcat(result, "|");
```
Comment (mfleisz) [inline libfreerdp/core/info.c]: Potential OOB write if all flags are set (strcat adds a terminating \0 to the end of the string).

File: client/common/cmdline.c [hunk 3]
```diff
@@ -1288,6 +1288,18 @@ static BOOL ends_with(const char* str, const char* ext)
 
 	return strncmp(&str[strLen - extLen], ext, extLen) == 0;
 }
+
+static void activate_smartcard_logon_rdp(rdpSettings* settings)
+{
+	settings->SmartcardLogon = TRUE;
+	settings->RdpSecurity = TRUE;
+	settings->TlsSecurity = FALSE;
```
Comment (hardening) [inline client/common/cmdline.c]: Just curious, isn't smartcard authentication possible with TLS security ?

File: libfreerdp/core/settings.c [hunk 4]
```diff
@@ -611,6 +611,7 @@ rdpSettings* freerdp_settings_new(DWORD flags)
 		goto out_fail;
 
 	settings->ActionScript = _strdup("~/.config/freerdp/action.sh");
+	settings->SmartcardLogon = FALSE;
 	return settings;
```
Comment (hardening) [inline libfreerdp/core/settings.c]: nitpicking but `settings` is calloced so SmartcardLogon is already `FALSE`

File: client/common/cmdline.c (hunk 3 above)
Comment (akallabeth) [reply client/common/cmdline.c]: @hardening Hmm, right. Should also be possible, will check.

**Pull request merged**